### PR TITLE
[AndroidToolTask] Log tool output as error 

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/xamarin-android-tools/issues/208

The `<AndroidToolTask/>` task has been updated to capture all output from
the tool it is running and log it as an error if the task fails.

All `<Exec/>` tasks that would run `adb` have been updated to use the
`<AndroidAdb/>` task instead to improve error handling and logging.

Bumps to xamarin/xamarin-android-tools@1ea4e35eaf

Changes: https://github.com/xamarin/xamarin-android-tools/compare/05f9a906379cabead66a07c378b37eda376597d2...1ea4e35eaf3fe09b9ca8adf0a29b4d9988146338
  * xamarin/xamarin-android-tools@1ea4e35: [AndroidToolTask] Log tool output as error
  * xamarin/xamarin-android-tools@c8a5b5b: Check for ANDROID_HOME Sdk location on Windows

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets [hunk 1]
```diff
@@ -55,17 +56,53 @@ This file contains targets specific for Android application projects.
...
+    />
 
     <Message Text="Running the application and waiting for it to fully start" Importance="High" />
-    <Exec Command="&quot;$(AdbToolPath)adb&quot; $(AdbTarget) shell am start -S -W -n &quot;$(_AndroidPackage)/$(AndroidLaunchActivity)&quot;" />
+    <AndroidAdb
+      ToolExe="$(AdbToolExe)"
+      ToolPath="$(AdbToolPath)"
+      AdbTarget="$(AdbTarget)"
+      Command="shell"
+      Arguments="-S -W -n &quot;$(_AndroidPackage)/$(AndroidLaunchActivity)&quot;"
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets:87]: I think this lost the `am start` portion of the command?

File: src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets [hunk 2]
```diff
@@ -55,17 +56,53 @@ This file contains targets specific for Android application projects.
...
+      AdbTarget="$(AdbTarget)"
+      Command="logcat"
+      Arguments="-d"
+      >
+      <Output TaskParameter="Result" PropertyName="_LogcatOutput" />
+    </AndroidAdb>
+    <WriteLinesToFile
+      File="$(_RunLogFilePath)"
+      Lines="$(_LogcatOutput)"
+    />
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets:105]: I would add `Overwrite="true"` (default is false), that should be the same behavior `>` had:

* https://learn.microsoft.com/en-us/visualstudio/msbuild/writelinestofile-task?view=vs-2022

File: src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets [hunk 3]
```diff
@@ -55,17 +56,53 @@ This file contains targets specific for Android application projects.
...
     <Message Text="Setting the debug.mono.log property to: $(_MonoLog)" Importance="High" />
-    <Exec Command="&quot;$(AdbToolPath)adb&quot; $(AdbTarget) shell setprop debug.mono.log $(_MonoLog)" />
-    <Exec Command="&quot;$(AdbToolPath)adb&quot; $(AdbTarget) logcat -G 16M" />
-    <Exec Command="&quot;$(AdbToolPath)adb&quot; $(AdbTarget) logcat -c" />
+    <AndroidAdb
+      ToolExe="$(AdbToolExe)"
+      ToolPath="$(AdbToolPath)"
+      AdbTarget="$(AdbTarget)"
+      Command="shell"
+      Arguments="debug.mono.log $(_MonoLog)"
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets:64]: we are missing the `setprop` bit here.

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets [hunk 4]
```diff
@@ -2583,9 +2583,6 @@ because xbuild doesn't support framework reference assemblies.
 
 <Target Name="_DeployApk"
     Condition=" '$(AndroidPackageFormat)' != 'aab' ">
-  <PropertyGroup>
-    <_DeployCommand>&quot;$(AdbToolPath)adb&quot; $(AdbTarget) install -r &quot;$(ApkFileSigned)&quot;</_DeployCommand>
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:2587]: why is this being removed? Its not mentioned in the PR description? 

I thought this and the _UnintallCommand were used by dotnet under the hood.

File: src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets (hunk 1 above)
Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets]: Thanks, updated

File: src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets (hunk 2 above)
Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets]: Thanks, updated

File: src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets (hunk 3 above)
Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets]: Thanks, updated

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets (hunk 4 above)
Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:2587]: I didn't see these `_DeployCommand` or `_UninstallCommand` command properties being used anywhere here or in monodroid. It looked like they were left overs from when the [`AndroidAdb` task was introduced](https://github.com/xamarin/xamarin-android/commit/1d7092d7b7382c62064eb45ccd70af5f34f62eba), and they could be safely removed?

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**